lib/autoupdate/rollout/strategy.go

```diff
@@ -40,28 +40,35 @@ const (
 // This interface allows us to inject dummy strategies for simpler testing.
 type rolloutStrategy interface {
 	name() string
-	progressRollout(context.Context, *autoupdate.AutoUpdateAgentRolloutStatus, time.Time) error
+	// progressRollout takes the new rollout spec, existing rollout status and current time.
+	// It updates the status resource in-place to progress the rollout to the next step if possible/needed.
+	progressRollout(context.Context, *autoupdate.AutoUpdateAgentRolloutSpec, *autoupdate.AutoUpdateAgentRolloutStatus, time.Time) error
 }
 
-func inWindow(group *autoupdate.AutoUpdateAgentRolloutStatusGroup, now time.Time) (bool, error) {
+func inWindow(group *autoupdate.AutoUpdateAgentRolloutStatusGroup, now time.Time, duration time.Duration) (bool, error) {
 	dayOK, err := canUpdateToday(group.ConfigDays, now)
 	if err != nil {
 		return false, trace.Wrap(err, "checking the day of the week")
 	}
 	if !dayOK {
 		return false, nil
 	}
-	return int(group.ConfigStartHour) == now.Hour(), nil
+
+	// We compute the theoretical window start and end
+	windowStart := now.Truncate(24 * time.Hour).Add(time.Duration(group.ConfigStartHour) * time.Hour)
+	windowEnd := windowStart.Add(duration)
+
+	return now.After(windowStart) && now.Before(windowEnd), nil
```

Comment:
The leading edge of the window is no longer in the window. Does that matter?

Reply:
Nor is the actual start? +1 to Stephen's question.

I suggest adding a comment saying what the intended interval is. For example, this seems to read like "(windowStart, windowEnd)" vs "[windowStart, windowEnd]".

Reply:
(Personally I tend to go for "[start,end)", as it's easy to chain sequences without overlap, but no strong opinions in this particular case.)

Reply:
TIl about the `[a,b)` open interval notation. French math use `[a,b[` to avoid the `(a,b)` ambiguity (open interval or pair).